self_hosted/c_output.oomph

```diff
@@ -544,44 +579,62 @@ class FilePair(
             assert(source.args.length() == 1)
             let c_path = "lib/generic/list.c"
             let h_path = "lib/generic/list.h"
-            let args = [new NamedType("ITEM", source.args.first())]
+            let itemtype = source.args.first()
+            let macrotypes = [
+                new NamedType("LIST", type),
+                new NamedType("ITEM", itemtype),
+            ]
```

Comment:
Use `Mapping` instead of `NamedType` for new code, even though #219 isn't merged yet